factfinder/main.py

```diff
@@ -226,65 +372,42 @@ def aggregate_horizontal(self, client, v, geotype) -> pd.DataFrame:
             )
-        return df[["census_geoid", "pff_variable", "geotype", "e", "m"]]
-
-    def aggregate_vertical(self, df, from_geotype, to_geotype):
-        """
-        this function will aggregate over geographies, 
-        e.g. aggregate over tracts to get NTA level data
-        """
-        options = {
-            "tract": {"NTA": tract_to_nta, "cd": tract_to_cd},
-            "block group": {
-                "cd_fp_500": block_group_to_cd_fp500,
-                "cd_fp_100": block_group_to_cd_fp100,
-                "cd_park_access": block_group_to_cd_park_access,
-            },
-            "block": {
-                "cd_fp_500": block_to_cd_fp500,
-                "cd_fp_100": block_to_cd_fp100,
-                "cd_park_access": block_to_cd_park_access,
-            },
-        }
-        return options[from_geotype][to_geotype](df)
+        return df
 
-    def download_variable(self, client, variables, geotype) -> pd.DataFrame:
+    def download_variable(self, client, v: Variable, geotype: str) -> pd.DataFrame:
         """
         Given a list of census_variables, and geotype, download data from acs api
         """
         geoqueries = self.get_geoquery(geotype)
-        _download = partial(self.download, client=client, variables=variables)
+        _download = partial(self.download, client=client, v=v)
         with Pool(5) as pool:
             dfs = pool.map(_download, geoqueries)
         df = pd.concat(dfs)
         return df
 
-    def download(self, geoquery, client, variables) -> pd.DataFrame:
+    def download(self, geoquery: list, client, v: Variable) -> pd.DataFrame:
         """
         this function works in conjunction with download_variable, 
         and is only created to facilitate multiprocessing
         """
-        return pd.DataFrame(
-            client.get(("NAME", ",".join(variables)), geoquery, year=self.year)
-        ).replace(
-            [
-                999999999,
-                333333333,
-                222222222,
-                666666666,
-                888888888,
-                555555555,
-                -999999999,
-                -333333333,
-                -222222222,
-                -666666666,
-                -888888888,
-                -555555555,
-            ],
-            np.nan,
+        # Create Variables
+        E_variables, M_variables = self.create_census_variables(v)
+
+        df = pd.DataFrame(
+            client.get(
+                ("NAME", ",".join(E_variables + M_variables)), geoquery, year=self.year
+            )
         )
 
-    def get_geoquery(self, geotype) -> list:
+        # If E is an outlier, then set M as Nan
+        for i in v.census_variable:
+            df.loc[df[f"{i}E"].isin(self.outliers), f"{i}M"] = np.nan
+
+        # Replace all outliers as Nan
+        df = df.replace(self.outliers, np.nan)
+        return df
+
+    def get_geoquery(self, geotype:str) -> list:
```

Comment:
Again an opportunity to switch away from if/elif, but might not be worth it

Reply:
this is a lower priority fix, nice to have